### DISCUSSION TITLE
FRRouting router implementation

### DISCUSSION THREAD

**Discussion by blaggacao** [nrdxp/nixdev#177] [Ideas]
## Ideas
Users of devos environments might also wish to configure their own router (e.g. a RPi) declaratively via this repository.

## Requests
I might be able to contribute a FRRouting package to the `community` branch with some basic configuration options so that users could easily configure their `./hosts/router.nix` with it.

/cc @DavHau would you be ok taking parts of your work as basis?

**Comment by blaggacao** [comment]
Note: [yang models](https://github.com/FRRouting/frr/tree/master/yang) seem to evolve as the nix-lang for the networking community. Example: https://github.com/FRRouting/frr/blob/master/yang/example/ripd.json

Might there be even room for a `yang.nix` project?

**Comment by DavHau** [comment]
> @DavHau would you be ok taking parts of your work as basis?

Of course. Just go ahead with whatever you can re-use

**Comment by blaggacao** [comment]
It seams feasible to simplify network management via use of ipv6 and link-local only typologies:

https://tools.ietf.org/html/rfc7404#section-2.2
